### PR TITLE
MINOR: Remove a couple of redundant `CoreUtils.rm` methods

### PR DESCRIPTION
**Pull request by ijuma**
Also:
- Rename remaining `CoreUtils.rm` to `delete` for consistency
- Use `try with resources` in `Utils` to simplify code
- Silence compiler warning due to exception catch clause in `TestUtils`

### CODE REVIEW

**Comment by ijuma** [comment]
Review by @guozhangwang please.

File: core/src/test/scala/unit/kafka/log/BrokerCompressionTest.scala [hunk 1]
```diff
@@ -26,9 +25,11 @@ import org.junit.Assert._
 import org.junit.runner.RunWith
 import org.junit.runners.Parameterized
 import org.junit.runners.Parameterized.Parameters
-import java.util.{Properties, Collection, ArrayList}
-import kafka.server.KafkaConfig
+import java.util.{Collection, Properties}
```
Comment (guozhangwang) [inline core/src/test/scala/unit/kafka/log/BrokerCompressionTest.scala]: nit: move this line after the kafka clients package imports?

File: core/src/test/scala/unit/kafka/server/HighwatermarkPersistenceTest.scala [hunk 2]
```diff
@@ -25,9 +26,10 @@ import org.junit._
 import org.junit.Assert._
 import kafka.common._
 import kafka.cluster.Replica
-import kafka.utils.{ZkUtils, SystemTime, KafkaScheduler, TestUtils, MockTime, CoreUtils}
+import kafka.utils.{CoreUtils, KafkaScheduler, MockTime, SystemTime, TestUtils, ZkUtils}
 import java.util.concurrent.atomic.AtomicBoolean
-import org.apache.kafka.common.utils.{MockTime => JMockTime}
+
+import org.apache.kafka.common.utils.{Utils, MockTime => JMockTime}
```
Comment (guozhangwang) [inline core/src/test/scala/unit/kafka/server/HighwatermarkPersistenceTest.scala]: nit: Move this import before java imports?

File: core/src/test/scala/unit/kafka/server/OffsetCommitTest.scala [hunk 3]
```diff
@@ -19,20 +19,19 @@ package kafka.server
...
 import kafka.utils.TestUtils._
 import kafka.zk.ZooKeeperTestHarness
 import org.apache.kafka.common.protocol.Errors
-
 import org.junit.{After, Before, Test}
-
 import java.util.Properties
 import java.io.File
 
+import org.apache.kafka.common.utils.Utils
```
Comment (guozhangwang) [inline core/src/test/scala/unit/kafka/server/OffsetCommitTest.scala]: nit: Move this line after line 26?

**Comment by guozhangwang** [comment]
LGTM except a few very minor comments.

**Comment by ijuma** [comment]
@guozhangwang Addressed the review comments. I wasn't aware that we group java.\* imports after other imports. Interesting, in my experience I've seen the opposite (java.\* imports first).

**Comment by guozhangwang** [comment]
@ijuma There are some discrepancies, yes. But the general idea is to group all kafka.\* imports, and all o.a.kafka.\* imports, and all common java / scala imports along with external dependencies like jackson / log4j.

**Comment by guozhangwang** [comment]
Merged to trunk and 0.10.0.

**Pull request closed without merging**